config/istio/generate.sh

```diff
@@ -14,5 +14,8 @@ fi
 echo "generating Istio resource definitions..." >&2
 istioctl manifest generate -f "${SCRIPT_DIR}/istio-values.yaml" "$@" | \
   ytt --ignore-unknown-comments \
+    --data-value istio_version=$DESIRED_ISTIO_VERSION \
+    -f "${SCRIPT_DIR}/values.yaml" \
     -f - \
-    -f "${SCRIPT_DIR}/overlays"
+    -f "${SCRIPT_DIR}/overlays" \
+    -f "${SCRIPT_DIR}/update-job" \
```

Comment:
Is there a reason to add this update-job to the generated istio manifest? It seems like it would be a better fit as a separate yaml file in our regular config directory.

Reply:
I put it in the same folder and build it with the istio configuration so that the istio version can get passed along with it. This makes updating the job for new istio versions into a side effect of updating the istio version itself, whereas putting it somewhere else I think we'd have to make a special effort to update it. I'm open to alternatives though.

Reply:
I see. Although I think the main problem I saw with this is that we can't easily change the workload-namespace when it is in this file, because we want to ytt parameterize it. Maybe it can live in the same directory, but not be ytt'd into the generated istio config.

Reply:
I was able to parameterize it by using the values file from the directory above. As part of doing so I had to add `istioVersion:` to that file. If we wanted, we could make this file the source of truth for the Istio version and have the generate script pull from it. How does that sound?

commit in question: https://github.com/cloudfoundry/cf-k8s-networking/commit/d4aff7abb82aafd9c8ff3640c0b94e4814527a4e